helix-rest/src/main/java/org/apache/helix/rest/client/CustomRestClientImpl.java

```diff
@@ -122,18 +144,23 @@ protected JsonNode getJsonObject(HttpResponse httpResponse) throws IOException {
 
   @VisibleForTesting
   protected HttpResponse post(String url, Map<String, String> payloads) throws IOException {
+    HttpPost postRequest = new HttpPost(url);
+    postRequest.setHeader("Accept", ACCEPT_CONTENT_TYPE);
+    StringEntity entity = new StringEntity(OBJECT_MAPPER.writeValueAsString(payloads), ContentType.APPLICATION_JSON);
+    postRequest.setEntity(entity);
+    HttpResponse httpResponse;
+    int hashCode = url.hashCode() + payloads.hashCode();
     try {
-      HttpPost postRequest = new HttpPost(url);
-      postRequest.setHeader("Accept", ACCEPT_CONTENT_TYPE);
-      StringEntity entity = new StringEntity(OBJECT_MAPPER.writeValueAsString(payloads),
-          ContentType.APPLICATION_JSON);
-      postRequest.setEntity(entity);
+      if (_cachedResults.containsKey(hashCode)) {
```

Comment:
Leave a TODO comments here. It is better to have response data processed after caching. Otherwise, every single call still needs to process the responses.

Reply:
It's a good suggestion. But keeping it HttpResponse is more general, gives us the ability to even add cache support for getInstance method. Converting from HttpResponse to output format is minor